### PR TITLE
feat(spanner): add NUMERIC support

### PR DESCRIPTION
**Pull request by skuruppu**

### CODE REVIEW

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs:205]: Consider using the `Value.ForString` helper method, designed to make `Value` easier to work with.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs:522]: Unwrap to one line? It's not terribly long.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs:211]: I think it's worth using ArgumentException here (as in the case above). And if we're supporting string as well, that should be in the exception message.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs
Comment (skuruppu) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs:205]: That's much nicer. Thank you.

**Comment by skuruppu** [comment]
Thanks for the super quick review @jskeet. I was mostly putting this PR up to make sure I was on the right track. We won't be able to merge this in for a few weeks since backend support is still not available. So I will put this back into draft mode. In the meantime I will look into the TODOs that I listed in the first comment.

**Comment by skuruppu** [comment]
@jskeet this is ready for another round of reviews. Since you last looked at it, I've added:
- Integration tests. Note that they can only run be against the staging backend for now.
- Type conversions from integral and floating-point numeric types to Numeric, and associated tests.

**Comment by jskeet** [comment]
I'm on vacation today, but I'll look at this and the integration-tests-against-emulator PR on Monday.

**Comment by skuruppu** [comment]
No rush @jskeet. Please take your time.

**Comment by skuruppu** [comment]
Sorry about that. The specific commit to review is https://github.com/googleapis/google-cloud-dotnet/pull/5057/commits/405dd3fbeebcafe6237c870adaaaa4a8904e0746. Nothing else has changed since you last reviewed the PR (sorry about the rebasing, I understand it makes code reviews difficult for incremental work).

If you're happy with the change, then after we merge this in, if you could help us cut a new release, that would be wonderful.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs:112]: (Not sure if the previous comment was actually registered)

Use Skip.If here?

**Review by jskeet — APPROVED** [review]
Thanks very much for the clarification. Happy to cut a release when it's all done.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs
Comment (skuruppu) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs:112]: Good catch. Done.

**Pull request merged**
